docs/reference/database-migration.md

```diff
@@ -0,0 +1,58 @@
+  * `Commit` A finishes, tries to update the `Branch` and fails due to the recent modification by `Commit` B,
+  * `Commit` A is retired, setting `StagingToken` to val3 and samples the `Branch`
+  * `Commit` B finishes, tries to update the `Branch` and fails due to the recent modification by `Commit` A,
+The first commit to succeed, releases this mutual interference, and the next commits can continue uninterrupted.
+
+Eventually , as retries will be exhausted, at least one commit will succeed, if not earlier, but this was definitely something that had to be taken into consideration when designing our KV Store. You can read more on the Commit Flow in the [dedicated section in the KV Design](https://github.com/treeverse/lakeFS/blob/master/design/accepted/metadata_kv/index.md#graveler-metadata---branches-and-staged-writes)
+
+### DB Transactions and Atomic Updates
+
+Another notable difference is the existence of DB transactions with PostgreSQL, ability that our KV Store lacks. This ability was leveraged by `lakeFS` to construct several DB updates, into one "atomic" operation - each failure, in each step, rolled back the entire operation, keeping the DB consistent and clean.
+With KV Store, this ability is gone, and we had to come up with various solutions. As a starting point, the DB consistency is, obviously, not anything we can risk. On the other hand, maintaining the DB clean, and as a result smaller, is something that can be sacrificed, at least as a first step. Let us take a look at a relatively simple flow of a new `Repository` creation:
+A brand new `Repository` has 3 objects: The `Repository` object itself, an initial `Branch` object and an initial `Commit`, which the `Branch` points to. With SQL DB, it was as simple as creating all 3 objects in the DB under one transaction (at this order). Any failure resulted in a rollback and no redundant leftovers in our DB.
+With no transaction in KV Store, if for example the `Branch` creation fails, it will leave the `Repository` without an initial `Branch` (or a `Branch` at all), yet the `Repository` will be accessible. Trying to delete the `Repository` as a response to `Branch` creation failure is ony a partial solution as this operation can fail as well.
+To mitigate this we introduced a per-`Repository`-partition, which holds all repository related objects (the `Branch` and `Commit` in this scenario). The partition key can only be derived from the specific`Repository` instance itself. In addition we first create the `Repository` objects, the `Commit` and the `Branch`, under the `Repository`'s partition key, and then the `Repository` is created. The `Repository` and its objects will be accessible only after a successful creation of all 3 entities. A failure in this flow might leave some dangling objects, but consistency is maintained.
+The amount of such dangling objects is not expected to be significant, and we plan to implement a cleaning algorithm to keep our KV Store neat and clean
```

Comment:
Consider removing this.  It depends on the workload.  Workloads that create many temporary objects where they write, or that write and then overwrite existing objects, will generate dangling objects.  It is hard for us to know what users run.

Reply:
@arielshaqed - this one is specifically reagrding a failure with repository creation, in which we can have a dangling branch and commit, at most. It also relies on the (maybe wrong) assumptions that repository creation is not something that users do periodically. 
Is that assumption wrong?
Do you still think we are better off without this last sentence?
Thanks

Reply:
Thanks for the explanation!  You're right (of course).